doc/nrf/libraries/bluetooth_services/bt_le_adv_prov.rst

```diff
@@ -0,0 +1,110 @@
+
+Providers
+=========
+
+The Bluetooth LE advertising providers subsystem gets both advertising data and scan response data from providers.
+A provider manages a single element that is part of either advertising data or scan response data.
+
+The |NCS| provides a set of predefined data providers.
+An application can also define application-specific data providers.
+
+Application-specific providers
+------------------------------
+
+An application-specific provider must implement :c:func:`bt_le_adv_prov_data_get` callback and register it using one of the following macros:
+
+* :c:macro:`BT_LE_ADV_PROV_AD_PROVIDER_REGISTER`
+  The macro registers provider that appends data to advertising packets.
+* :c:macro:`BT_LE_ADV_PROV_SD_PROVIDER_REGISTER`
+  The macro registers provider that appends data to scan response packets.
+
+When the callback is called, the provider must fill in the Bluetooth data structure passed as a callback argument.
+Apart from filling the Bluetooth data structure, the provider can also perform any of the following actions:
+
+* Fill out the feedback structure (:c:struct:`bt_le_adv_prov_feedback`) to provide additional information to the module that controls Bluetooth LE advertising.
+* Return an error code.
+  The error value of ``-ENOENT`` is used to inform that provider to desists from providing data.
+  In that case, the provider does not fill in the Bluetooth data structure.
+  This error code is not forwarded to the module that controls Bluetooth LE advertising.
+  Other negative values denote provider-specific errors, which are forwarded directly to the module that controls Bluetooth LE advertising.
+
+The provided Bluetooth data can depend on any of the following options:
+
+* Bluetooth advertising state passed as function argument (:c:struct:`bt_le_adv_prov_adv_state`).
+* Calls of the provider's dedicated API, if the provider exposes such an API.
+
+For example, a provider may fill in the Bluetooth data if it is not bonded.
+The provider returns ``-ENOENT`` to desist from providing data if bonded.
+Examples of provider implementations can be found in the :file:`subsys/bluetooth/bt_le_adv_prov/providers/` file.
+
+Configuration
+*************
+
+Set :kconfig:option:`CONFIG_BT_LE_ADV_PROV` to enable the Bluetooth LE advertising providers subsystem.
+
+Predefined providers
+====================
+
+The |NCS| provides a set of predefined providers.
+Each provider is enabled using a dedicated Kconfig option.
+These options share common prefix of ``CONFIG_BT_LE_ADV_PROV_``.
+
+Among others, the following providers are available:
+
+* Advertising Flags (:kconfig:option:`CONFIG_BT_LE_ADV_PROV_FLAGS`),
+* GAP Appearance (:kconfig:option:`CONFIG_BT_LE_ADV_PROV_GAP_APPEARANCE`),
+* Microsoft Swift Pair (:kconfig:option:`CONFIG_BT_LE_ADV_PROV_SWIFT_PAIR`),
+* TX Power (:kconfig:option:`CONFIG_BT_LE_ADV_PROV_TX_POWER`).
+
+For details about given advertising provider, see Kconfig help.
```

Comment:
```suggestion
For details about each advertising provider, see the Kconfig option description.
```
That's the help you mean, right?

Reply:
In Kconfig it's defined as `help`, but term description should not cause confusion here I think